components/rhs_thread/rhs_thread.tsx

```diff
@@ -102,6 +102,14 @@ export default class RhsThread extends React.Component<Props, State> {
         this.rhspostlistRef = React.createRef();
     }
 
+    public getSnapshotBeforeUpdate(prevProps: Props) {
+        if (prevProps.posts.length < this.props.posts.length) {
+            const list = this.rhspostlistRef.current;
+            return list ? list.scrollHeight - list.scrollTop : null;
```

Comment:
I think it might be a bit nicer to return an object like `{atBottom: true}` here instead of just returning a number.

Reply:
Also, it looks like the type of the snapshot is the 3rd type parameter like `RhsThread extends React.PureComponent<Props, State, Snapshot>`